chef/cookbooks/provisioner/templates/suse/crowbar_register.erb

```diff
@@ -46,9 +46,16 @@ while test $# -gt 0; do
         -h|--help|--usage|-\?) usage ;;
         -v|--verbose) CROWBAR_VERBOSE=1 ;;
         -f|--force) CROWBAR_FORCE=1 ;;
-        -k|--keep-existing-hostname) KEEP_EXISTING_HOSTNAME=1 ;;
         --gpg-auto-import-keys) CROWBAR_AUTO_IMPORT_KEYS=1 ;;
         --no-gpg-checks) CROWBAR_NO_GPG_CHECKS=1 ;;
+        -k|--keep-existing-hostname)
+            if test $# -eq 1; then
```

Comment:
> We require only a parameter for -k|--keep-existing-hostname where the option is doesn't matter.

How is that? Maybe I am also missreading the code. But if e.g. somebody calls ```crowbar_register -k -v``` (which is missing the 1/0 parameter for -k)  ```test $# -eq 1``` is false and ```KEEP_EXISTING_HOSTNAME``` will be set to ```-v```. I guess this is not what the code should do, right? Or am I missing something?